### PR TITLE
511 material properties add support for bounds and physical bounds on the output of material laws

### PR DESCRIPTION
**Pull request by thelfer**

### CODE REVIEW

File: mfront/src/GenericMaterialPropertyInterfaceBase.cxx [hunk 1]
```diff
@@ -175,7 +175,8 @@ namespace mfront {
          << "mfront_output_status->status = -1;\n"
          << "mfront_output_status->bounds_status = -" << i << ";\n"
          << "errno = mfront_errno_old;\n"
-         << "return nan(\"" << v.name << " is out of bounds.\");\n"
+         //         << "return nan(\"" << v.name << " is out of bounds.\");\n"
```
Comment (maxence-wz) [inline mfront/src/GenericMaterialPropertyInterfaceBase.cxx:178]: to be removed

File: mfront/src/GenericMaterialPropertyInterfaceBase.cxx [hunk 2]
```diff
@@ -191,8 +192,10 @@ namespace mfront {
          << "mfront_report(\"" << v.name << " is over its upper bound (\" + "
          << to_string << " + \">" << b.upperBound << ").\\n\");\n"
          << "mfront_output_status->status = -1;\n"
-         << "mfront_output_status->bounds_status = -" << i << ";\n"
-         << "return nan(\"" << v.name << " is out of bounds.\");\n"
+         << "mfront_output_status->bounds_status = -" << i
+         << ";\n"
+         //         << "return nan(\"" << v.name << " is out of bounds.\");\n"
```
Comment (maxence-wz) [inline mfront/src/GenericMaterialPropertyInterfaceBase.cxx:197]: to be removed

File: mfront/src/GenericMaterialPropertyInterfaceBase.cxx [hunk 3]
```diff
@@ -216,7 +219,8 @@ namespace mfront {
          << "mfront_output_status->status = -1;\n"
          << "mfront_output_status->bounds_status = -" << i << ";\n"
          << "errno = mfront_errno_old;\n"
-         << "return nan(\"" << v.name << " is out of bounds.\");\n"
+         //         << "return nan(\"" << v.name << " is out of bounds.\");\n"
```
Comment (maxence-wz) [inline mfront/src/GenericMaterialPropertyInterfaceBase.cxx:222]: to be removed

File: mfront/include/MFront/CMaterialPropertyInterfaceBase.hxx [hunk 4]
```diff
@@ -264,10 +264,10 @@ namespace mfront {
     /*!
      * \brief write catch blocks before the material property body
      * \param[out] os:  output file stream
-     * \param[in] msh: code returning the error message (a string must be escaped)
-     * \param[in] mpd: material property description
-     * \param[in] floating_point_type: floating-point type
-     * \param[in] use_qt: use quantities
+     * \param[in] msh: code returning the error message (a string must be
+     * escaped) \param[in] mpd: material property description \param[in]
+     * floating_point_type: floating-point type \param[in] use_qt: use
```
[the remark this answers is not in the document]
Comment (thelfer) [reply mfront/include/MFront/CMaterialPropertyInterfaceBase.hxx:269]: done

File: mfront/src/GenericMaterialPropertyInterfaceBase.cxx (hunk 1 above)
Comment (thelfer) [reply mfront/src/GenericMaterialPropertyInterfaceBase.cxx:178]: done

File: mfront/src/GenericMaterialPropertyInterfaceBase.cxx (hunk 2 above)
Comment (thelfer) [reply mfront/src/GenericMaterialPropertyInterfaceBase.cxx:197]: done

**Pull request merged**